lib/urn.js

```diff
@@ -44,7 +44,77 @@ module.exports = {
       if (!urn.match(allowedProtocolsRegExp) || !urn.match(/^([^:]+:)+[^:]+$/)) {
         return false;
       }
+
+      // Parse ID portion of URN
+      var cat = urn.split(':').slice(0, -1)
+      var id = urn.split(':').slice(-1)[0];
+
+      // Split identity by each byte
+      var charArr = id.match(/.{1,2}/g);
```

Comment:
Yea, this simply breaks each string by every 2 characters. Then I can validate based on the leading byte `02,03,04` and do extra validation.